Code/Framework/AzFramework/AzFramework/Quality/QualityCVarGroup.cpp

```diff
@@ -0,0 +1,205 @@
+        {
+            // optional default quality level
+            settingsRegistry->GetObject(m_qualityLevel, FixedValueString::format("%.*s/Default", AZ_STRING_ARG(path)));
+
+            // optional description
+            settingsRegistry->Get(description, FixedValueString::format("%.*s/Description", AZ_STRING_ARG(path)));
+        }
+
+        m_functor = AZStd::make_unique<QualityCVarGroupFunctor>(
+            m_name.c_str(),
+            description.c_str(),
+            AZ::ConsoleFunctorFlags::DontReplicate,
+            AZ::AzTypeInfo<ValueType>::Uuid(),
+            *this,
+            &QualityCVarGroup::CvarFunctor);
+    }
+
+    QualityCVarGroup::~QualityCVarGroup() = default;
+
+    inline void QualityCVarGroup::CvarFunctor(const AZ::ConsoleCommandContainer& arguments)
+    {
+        StringToValue(arguments);
+    }
+
+    void QualityCVarGroup::LoadQualityLevel(ValueType qualityLevel)
+    {
+        using FixedValueString = AZ::SettingsRegistryInterface::FixedValueString;
+        using VisitResponse = AZ::SettingsRegistryInterface::VisitResponse;
+        using Type = AZ::SettingsRegistryInterface::Type;
+
+        auto settingsRegistry = AZ::SettingsRegistry::Get();
+        if (settingsRegistry == nullptr)
+        {
+            return;
+        }
+
+        auto key = FixedValueString::format("%s/Settings", m_path.c_str());
+        auto settingsVisitorCallback = [&, settingsRegistry](const AZ::SettingsRegistryInterface::VisitArgs& visitArgs)
+        {
+            AZStd::string_view command = visitArgs.m_fieldName;
+
+            if (visitArgs.m_type == Type::Object ||
+                visitArgs.m_type == Type::Null ||
+                visitArgs.m_type == Type::NoType)
+            {
+                AZ_Warning("QualityCVarGroup", false, "Invalid setting value type for %.*s, objects and null are not supported, only arrays, bool, string, or numbers.", AZ_STRING_ARG(command));
+                return VisitResponse::Skip;
+            }
+
+            AZ::PerformCommandResult result{};
+            if (visitArgs.m_type == Type::Array)
+            {
+                AzFramework::QualityLevels currentQualityLevel{ 0 };
+                AZStd::string_view arrayValueKey = visitArgs.m_jsonKeyPath;
+                auto getHighestQualityLevel = [&](const AZ::SettingsRegistryInterface::VisitArgs& visitArrayArg)
+                {
+                    if (currentQualityLevel == qualityLevel)
+                    {
+                        result = PerformConsoleCommand(command, visitArrayArg.m_jsonKeyPath);
+                        return VisitResponse::Done;
+                    }
+                    currentQualityLevel++;
+                    return VisitResponse::Continue;
+                };
+                AZ::SettingsRegistryVisitorUtils::VisitArray(*settingsRegistry, getHighestQualityLevel, visitArgs.m_jsonKeyPath);
+                if (currentQualityLevel == AzFramework::QualityLevels(0))

```

Comment:
This doesn't need to cast `0` to the QualityLevels enum and can just use the DefaultQualityLevel option
Also the `AzFramework::` namespace doesn't need to be specified.
```suggestion
                if (currentQualityLevel == QualityLevels::DefaultQualityLevel)

```

Reply:
Thanks, good use of `DefaultQualityLevel`